internal/plugin/resource_plugin.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package plugin
+
+import (
+	"context"
+	"log/slog"
+	"sync"
+
+	"github.com/nginx/agent/v3/api/grpc/mpi/v1"
+	"github.com/nginx/agent/v3/internal/bus"
+	"github.com/nginx/agent/v3/internal/service"
+)
+
+type Resource struct {
+	messagePipe     bus.MessagePipeInterface
+	resourceService service.ResourceServiceInterface
+	resource        *v1.Resource
+	resourceMutex   sync.Mutex
+}
+
+func NewResource() *Resource {
+	return &Resource{
+		resourceMutex:   sync.Mutex{},
+		resourceService: service.NewResourceService(),
+		resource:        &v1.Resource{},
```

Comment:
Not sure about prepending 'resource' to everything inside the `Resource` struct, seems a bit redundant.

Reply:
removed resource plugin as it isn't needed